libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -405,6 +405,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 // return the actionResult as the result of this operation
                 return await dc.EndDialogAsync(result: requestResult, cancellationToken: cancellationToken).ConfigureAwait(false);
             }
+            catch (HttpRequestException ex)
+            {
+                var requestResult = new Result()
+                {
+                    StatusCode = (int)HttpStatusCode.NotFound,
+                    ReasonPhrase = ex.Message
+                };
+
+                if (ResultProperty != null)
+                {
+                    dc.State.SetValue(ResultProperty.GetValue(dc.State), requestResult);
+                }
+
+                traceInfo.response = JObject.FromObject(requestResult);
+
+                // Write Trace Activity for the http request and response values
+                await dc.Context.TraceActivityAsync("HttpRequest", (object)traceInfo, valueType: "Microsoft.HttpRequest", label: Id).ConfigureAwait(false);
```

Comment:
nit: see next comment... this line is also common to the try and catch blocks so could be outside of both, or maybe a helper function